File: frontend/src/api/publicPlaces.js
import apiClient from "./client";

export const getNearbyPublicPlaces = (lat, lng, radiusKm = 5) =>
  apiClient
    .get("/api/public-places/nearby", { params: { lat, lng, radius_km: radiusKm } })
    .then((r) => r.data);

File: backend/app/routers/public_places.py
from fastapi import APIRouter, Query

from app.schemas.public_place import PublicPlaceResponse
from app.services.public_places import fetch_nearby_public_places

router = APIRouter(prefix="/api/public-places", tags=["public-places"])


@router.get("/nearby", response_model=list[PublicPlaceResponse])
def nearby_public_places(
    lat: float = Query(...),
    lng: float = Query(...),
    radius_km: float = Query(5.0, gt=0, le=20),
):
    return fetch_nearby_public_places(lat, lng, radius_km)
